src/main/java/net/wouterdanes/docker/maven/AbstractDockerMojo.java

```diff
@@ -114,6 +114,29 @@ protected void registerBuiltImage(String imageId, ImageBuildConfiguration imageC
         return Collections.unmodifiableCollection(builtImagesMap.values());
     }
 
+    protected void cleanUpStartedContainers() {
+        List<String> stoppedContainerIds = new ArrayList<>();
+        for (Iterator<StartedContainerInfo> it = getStartedContainers().iterator(); it.hasNext();) {
+            String containerId = it.next().getContainerInfo().getId();
+            getLog().info(String.format("Stopping container '%s'..", containerId));
+            try {
+                getDockerProvider().stopContainer(containerId);
+                it.remove();
+                stoppedContainerIds.add(containerId);
+            } catch (DockerException e) {
+                getLog().error("Failed to stop container (this means it also won't be deleted)", e);
+            }
+        }
+        for (String containerId : stoppedContainerIds) {
```

Comment:
Note this is a change in behaviour (arguably a bug fix): original behaviour was to try and delete containers which failed to be stopped, which doesn't work if the container is still running unless the delete call also includes `force=true` (equivalent to `docker rm -f`).